src/modules/silentpayments/main_impl.h

```diff
@@ -0,0 +1,797 @@
+            secp256k1_scalar_clear(&output_tweak_scalar);
+            secp256k1_memclear_explicit(&shared_secret, sizeof(shared_secret));
+            return 0;
+        }
+
+        /* Calculate output = spend_pubkey + output_tweak * G.
+         * This can fail if output_tweak * G is the negation of spend_pubkey, but this happens only
+         * with negligible probability for honestly created spend_pubkey as output_tweak is the output of a hash function. */
+        if (!secp256k1_eckey_pubkey_tweak_add(&output_ge, &output_tweak_scalar)) {
+            /* Leaking these values would break indistinguishability of the transaction, so clear them. */
+            secp256k1_scalar_clear(&output_tweak_scalar);
+            secp256k1_memclear_explicit(&shared_secret, sizeof(shared_secret));
+            return 0;
+        }
+        found = 0;
+        secp256k1_xonly_pubkey_save(&output_xonly, &output_ge);
+        for (j = 0; j < n_tx_outputs; j++) {
+            if (secp256k1_xonly_pubkey_cmp(ctx, &output_xonly, tx_outputs[j]) == 0) {
+                label_tweak = NULL;
+                found = 1;
+                found_idx = j;
+                break;
+            }
+
+            /* If not found, proceed to check for labels (if a label lookup function is provided). */
+            if (label_lookup != NULL) {
+                secp256k1_ge output_negated_ge, tx_output_ge;
+                secp256k1_gej tx_output_gej, label_gej;
+                unsigned char label33[33];
+                size_t len;
+
+                secp256k1_xonly_pubkey_load(ctx, &tx_output_ge, tx_outputs[j]);
+                secp256k1_gej_set_ge(&tx_output_gej, &tx_output_ge);
+                secp256k1_ge_neg(&output_negated_ge, &output_ge);
+                secp256k1_gej_add_ge_var(&label_gej, &tx_output_gej, &output_negated_ge, NULL);
+                secp256k1_ge_set_gej_var(&label_ge, &label_gej);
+                ret = secp256k1_eckey_pubkey_serialize(&label_ge, label33, &len, 1);
+                /* Serialize must succeed because the point was just loaded.
+                 *
+                 * Note: serialize will also fail if label_ge is the point at infinity, but we know
+                 * this cannot happen since we only hit this branch if tx_output != output_xonly.
+                 * Thus, we know that label_ge = tx_output_gej + output_negated_ge cannot be the
+                 * point at infinity.
+                 */
+                VERIFY_CHECK(ret && len == 33);
+                label_tweak = label_lookup(label33, label_context);
+                if (label_tweak != NULL) {
+                    found = 1;
+                    found_idx = j;
+                    break;
+                }
+
+                secp256k1_gej_neg(&label_gej, &tx_output_gej);
+                /* If not found, negate the tx_output and calculate second scan label candidate:
+                 *     label2 = -tx_output - generated_output
+                 */
```

Comment:
This is useful and we had a similar line for label1 (`label1 = tx_output - generated_output`). Did you remove that intentionally?